### ISSUE TITLE
Cheat codes: Re-evaluate button sequence on fail

### ISSUE THREAD

**Issue by Brycey92** [AEFeinstein/Super-2023-Swadge-FW#367]
Currently, if you press `UP` 3 times in either the main menu or Swadge Land, you will be placed in a clean state of cheat code entry, rather than at the point of having entered the first 2 buttons successfully.

**Comment by AEFeinstein** [comment]
Yep, that makes sense. Prior year's cheat codes bitshifted button states into a byte, then compared that byte for the unlock code, so it was more of a sliding window than an exact sequence.

**Comment by Brycey92** [comment]
Hey, that's not a bad idea!
